test/functional/tests/lazy_writes/recovery/test_recovery_flush_reset.py

```diff
@@ -0,0 +1,193 @@
+        core_device = core_disk.partitions[0]
+        core_device_link = core_device.get_device_link("/dev/disk/by-id")
+        cache_device_link = cache_device.get_device_link("/dev/disk/by-id")
+
+    with TestRun.step("Create test files."):
+        source_file, target_file = create_test_files(test_file_size)
+
+    with TestRun.step("Setup cache and add core."):
+        cache = casadm.start_cache(cache_device, cache_mode)
+        core = cache.add_core(core_device)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Copy file to CAS."):
+        copy_file(source=source_file.full_path, target=core.system_path, size=test_file_size,
+                  direct="oflag")
+
+    with TestRun.step("Sync and flush buffers."):
+        os_utils.sync()
+        output = TestRun.executor.run(f"hdparm -f {core.system_path}")
+        if output.exit_code != 0:
+            raise CmdException("Error during hdparm", output)
+
+    with TestRun.step("Trigger flush."):
+        TestRun.executor.run_in_background(cli.flush_cache_cmd(f"{cache.cache_id}"))
+
+    with TestRun.step("Hard reset DUT during data flushing."):
+        power_cycle_dut(wait_for_flush_begin=True, core_device=core_device)
+        cache_device.full_path = cache_device_link.get_target()
+        core_device.full_path = core_device_link.get_target()
+
+    with TestRun.step("Copy file from core and check if current md5sum is different than "
+                      "before restart."):
+        copy_file(source=core_device_link.get_target(), target=target_file.full_path,
+                  size=test_file_size, direct="iflag")
+        compare_files(source_file, target_file, should_differ=True)
+
+    with TestRun.step("Load cache."):
+        cache = casadm.load_cache(cache_device)
+        if cache.get_dirty_blocks() == Size.zero():
+            TestRun.fail("There are no dirty blocks on cache device.")
+
+    with TestRun.step("Stop cache with dirty data flush."):
+        core_writes_before = core_device.get_io_stats().sectors_written
+        cache.stop()
+        if core_writes_before >= core_device.get_io_stats().sectors_written:
+            TestRun.fail("No data was flushed after stopping cache started with load option.")
+
+    with TestRun.step("Copy test file from core device to temporary location. "
+                      "Compare it with the first version – they should be the same."):
+        copy_file(source=core_device_link.get_target(), target=target_file.full_path,
+                  size=test_file_size, direct="iflag")
+        compare_files(source_file, target_file)
+
+    with TestRun.step("Cleanup core device and remove test files."):
+        Dd() \
+            .input("/dev/zero") \
+            .output(core_device.system_path) \
+            .block_size(Size(1, Unit.Blocks4096)) \
+            .count(10) \
+            .oflag("direct") \
+            .run()
```

Comment:
Why is this needed?

Reply:
At the beginning I had more than one iteration in this test so I had to clean core before next iteration, but I changed it to parametrize, so it can be removed now